core/governance/engine.go

```diff
@@ -368,9 +369,25 @@ func (e *Engine) OnTick(ctx context.Context, t time.Time) ([]*ToEnact, []*VoteCl
 		e.gss.changedNodeValidation = true
 	}
 
-	for _, ep := range toBeEnacted {
+	for i, ep := range toBeEnacted {
 		// this is the new market proposal, and should already be in the slice
 		prop := *ep.ProposalData()
+
+		mktState, err := e.markets.GetMarketState(prop.ID)
+		if err != nil {
+			e.log.Error("could not get state of market %s", logging.String("market-id", prop.ID))
+			continue
+		}
+		if mktState == types.MarketStateCancelled {
+			toBeRemoved = append(toBeRemoved, prop.ID)
```

Comment:
I think this has no effect because the removal already takes place earlier in line 340 in the old code

Reply:
I believe calling the removal on each element in the `toBeRemoved` list was done above, because there were no new additions to in until the end.
I moved it further now here https://github.com/vegaprotocol/vega/pull/5668/commits/a11e922add05a5659da732ef070f6e0cc73cb00f#diff-dedf0633c5a9f40bc6bd64eda022d287b45a334e77b08ca25c307069777582b5R408
Thanks!